Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -140,6 +185,10 @@ private void CacheJobBinaries(JobBase job, IJobLogger logger)
                     UpdateAppConfigs(tempJobInstancePath);
 
                     _workingDirectory = tempJobInstancePath;
+
+                    // cache the file map snapshot for next time (to aid in detecting
+                    // file deletions)
+                    _cachedSourceDirectoryFileMap = sourceDirectoryFileMap;
```

Comment:
_workingDirectory is only set after all the files are copied to the cache dir, right? Before that it is null, and WorkingDirectory is null. So from what I see in the code, when a job is initialized, WorkingDirectory is null until the initial copy is complete. Can you please come over to my desk so we can discuss? :)